File: Easy/Javascript/231. Power of Two.js
/**Given an integer n, return true if it is a power of two. Otherwise, return false.
 * An integer n is a power of two, if there exists an integer x such that n == 2x.
 */

/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfTwo = function(n) {
    let power = 0;

    while (true){
        let val = 2 ** power;
        if (val >= n){
            return val === n ? true : false
        }
        power += 1
    }
};

console.log(isPowerOfTwo(32))

File: Easy/Python/231. Power of Two.py
'''
Given an integer n, return true if it is a power of two. Otherwise, return false.

An integer n is a power of two, if there exists an integer x such that n == 2x.
'''

class Solution:
    def isPowerOfTwo(self, n: int, powered=0) -> bool:
        val = 2 ** powered
        if val >= n:
            return n == val
        return self.isPowerOfTwo(n, powered+1)

print(Solution().isPowerOfTwo(5))